### PR TITLE
Add exclude and grep functions

### PR DESCRIPTION
**Pull request by shanson7**

### CODE REVIEW

File: expr/func_grep.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		return nil, err
+	}
+
+	var outputs []models.Series
+	for _, serie := range series {
+		if s.pattern.MatchString(serie.Target) != s.excludeMatches {
+			outputs = append(outputs, serie)
+		}
+	}
+	return outputs, nil
```
Comment (Spellchaser) [inline expr/func_grep.go]: Is it okay to return an empty slice? Should we provide a warning if it's empty?

File: expr/func_grep_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		pattern    string
+		in         []string
+		matches    []string
+		nonmatches []string
+	}{
+		{
+			"this",
+			[]string{"series.name.this.ok"},
+			[]string{"series.name.this.ok"},
+			[]string{},
```
Comment (Spellchaser) [inline expr/func_grep_test.go]: On the same idea, do we need a test to specify what happens when grep finds nothing?

File: expr/func_grep.go (hunk 1 above)
Comment (shanson7) [reply expr/func_grep.go]: Yes, an empty slice is fine. No need to warn.

File: expr/func_grep_test.go (hunk 2 above)
Comment (shanson7) [reply expr/func_grep_test.go]: This test has an "exclude" that excludes everything. It's the same code at that point.

File: expr/func_grep.go (hunk 1 above)
Comment (Spellchaser) [reply expr/func_grep.go]: I looked around guess convention is to send back empty. (y)

**Comment by Spellchaser** [comment]
lgtm, I wonder if there'll be problems with regex escaping or specialized names, but I don't know enough about the project :P

**Pull request merged**